### PR TITLE
refactor(components)!: move components out to go into @yext/pages-components

### PR DESCRIPTION
**Pull request by shrishayap**
Deleted everything in `pages/src/components` because it was moved to a separate repo and will be packaged as `@yext/pages-components`. Saw that there were no dependencies within the `pages` repo for components. Tested by packing a local version of this, and packing a local version of the new `@yext/pages-components` package, importing them into a local test repo, and seeing that components rendered properly (except for `<LocationMap>` but that never renders properly when local packing package).

### CODE REVIEW

**Comment by shrishayap** [comment]
Should I also update all the READMEs that say Components live here?

**Comment by mkilpatrick** [comment]
> Should I also update all the READMEs that say Components live here?

That'd be great.

File: packages/pages/package.json [hunk 1]
```diff
@@ -12,8 +12,7 @@
   "exports": {
     ".": "./dist/index.js",
     "./vite-plugin": "./dist/vite-plugin/plugin.js",
-    "./util": "./dist/util/index.js",
-    "./components": "./dist/components/index.js"
+    "./util": "./dist/util/index.js"
```
Comment (mkilpatrick) [inline packages/pages/package.json:15]: A lot of dependencies can be removed now.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -44,7 +44,7 @@ In most circumstance all changes should be associate with a unit test that lives
 - PR titles must follow the [semantic pull request](https://github.com/marketplace/actions/semantic-pull-request) style so that changelogs can be automatically generated. PR titles are automatically validated by a Github action.
 
   - ```
-      feat(components): add `Button` component.
+      feat(dev): add hydration option
```
Comment (mkilpatrick) [inline CONTRIBUTING.md:47]: Update the available scopes in the .github/ action

File: .eslintrc.json [hunk 3]
```diff
@@ -20,9 +20,6 @@
   "plugins": ["react", "@typescript-eslint", "import"],
   "rules": {
     "@typescript-eslint/no-explicit-any": "off",
-    "@typescript-eslint/no-empty-function": "off",
```
Comment (mkilpatrick) [inline .eslintrc.json:23]: Don't remove these

File: packages/pages/package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@yext/pages",
-  "version": "1.0.0-beta.26",
+  "version": "1.0.0-beta.23",
```
Comment (mkilpatrick) [inline packages/pages/package.json:3]: You don't want to downgrade this

File: packages/pages/package.json [hunk 5]
```diff
@@ -57,122 +51,108 @@
...
-    "esbuild": "^0.18.11",
-    "generate-license-file": "^2.0.0",
-    "glob": "^10.3.1",
-    "identity-obj-proxy": "^3.0.0",
-    "jest": "^29.6.0",
-    "jest-environment-jsdom": "^29.6.0",
-    "minimatch": "^9.0.2",
-    "ts-jest": "^29.1.1",
+    "@types/react": "^17.0.49",
+    "@types/react-dom": "^17.0.17",
```
Comment (mkilpatrick) [inline packages/pages/package.json:98]: All of these packages are now downgraded

File: packages/pages/package.json [hunk 6]
```diff
@@ -57,122 +51,108 @@
...
+    "minimatch": "^5.1.0",
+    "react": "^17.0.2",
+    "react-dom": "^17.0.2",
+    "ts-jest": "^28.0.8",
     "ts-jest-mock-import-meta": "^1.0.0"
   },
   "peerDependencies": {
-    "react": "^17.0.2 || ^18.2.0",
-    "react-dom": "^17.0.2 || ^18.2.0"
+    "react": "^17.0.2",
```
Comment (mkilpatrick) [inline packages/pages/package.json:114]: Needs 18 support as an option

File: packages/pages/package.json [hunk 7]
```diff
@@ -57,122 +51,108 @@
...
+    "react-dom": "^17.0.2"
   },
   "jest": {
     "rootDir": "..",
     "collectCoverageFrom": [
       "<rootDir>/pages/src/**/*.{ts,tsx,js}"
     ],
+    "globals": {
```
Comment (mkilpatrick) [inline packages/pages/package.json:122]: This is a regression. Needs to be moved under transform.

File: packages/pages/src/bin/spawn.ts [hunk 8]
```diff
@@ -13,38 +13,10 @@ const pathToPagesScript = path.resolve(
...
-  "loader.js"
-);
-
-const nodeVersion = Number(
-  spawnSync("node", ["-v"], { encoding: "utf-8" })
-    .stdout.substring(1)
-    .split(".")[0]
-);
-
-const experimentalFlags = ["--experimental-vm-modules"];
```
Comment (mkilpatrick) [inline packages/pages/src/bin/spawn.ts:32]: Need all of this. It was part of the support React 18 PR.

File: packages/pages/src/build/build.ts [hunk 9]
```diff
@@ -1,20 +1,26 @@
-import { Command } from "commander";
+import { CommandModule } from "yargs";
```
Comment (mkilpatrick) [inline packages/pages/src/build/build.ts:1]: Commander is what we changed to from yargs. I think you'll want to redo your merge. Looks like you reverted a lot of things.

**Comment by shrishayap** [comment]
Put up new PR

**Pull request closed without merging**